fbw-a380x/src/systems/instruments/src/SD/SystemDisplay.tsx

```diff
@@ -132,17 +137,28 @@ export const SystemDisplay = () => {
   };
 
   const updateCallback = (deltaTime) => {
+    if (ecamButtonLightDelayTimer != Number.MIN_SAFE_INTEGER) {
+      setEcamButtonLightDelayTimer((t) => t - deltaTime);
+      if (ecamButtonLightDelayTimer <= 0) {
+        setPage(currentPage);
+        setEcamButtonLightDelayTimer(Number.MIN_SAFE_INTEGER);
+      }
+    }
     if (ecamAllButtonPushed && !prevEcamAllButtonState) {
       // button press
-      setPage((prev) => (prev + 1) % 12);
-      setCurrentPage((prev) => (prev + 1) % 12);
+      setCurrentPage((prev) => {
+        prev = page === SdPages.None ? SdPages.Eng : prev + 1;
+        setEcamButtonLightDelayTimer(ECAM_LIGHT_DELAY_ALL);
+        return prev % SdPages.Crz; // CRZ page should not be shown when pressing ALL button - return 0
```

Comment:
Suggestion: it would be a lot safer to use an array (private static class member) with the enum values that should be cycled, then you can do arithmetic on the array indices and length (rather than assuming that SdPages.Cb is last for example). Treating an enum as an int is not really type safe even if it is regrettably allowed by typescript (that is the reason most typescript guides will recommend always using string enums).

Reply:
We had this discussion on Discord yesterday. 
Using an enum which is explicitly defined as numbers is a perfectly good pattern. It is much better than a magic number as it has been before which in fact tripped me up (suddenly there is an 11 or 12 in the code).

Regarding the algorithm; I used what has been done in the A32NX. This is a fix PR and not a rewrite of the code. 
(Also personally I think is it think this is actually easier to understand than introducing a new array just for the cycling.)

Reply:
Relying on implementation details is not a good pattern. We should just use an array so that the order is decoupled to whatever values the enum has.

Reply:
Go ahead ... I just used what was there already. I was trying to fix an issue on the A380X not rewrite someone else's code. 

(I still have a different opinion but really no energy to debate you guys on this)